src/services/ClientService.ts

```diff
@@ -316,6 +318,26 @@ class ClientService {
     return newTicket;
   }
 
+  static async getTicket(ticketId: ServiceRequestType['id']) {
+    const { data: ticket, error } = await supabaseClient
+      .from('service_requests')
```

Comment:
I politely disagree here. A join would reduce the number of DB queries/network calls, but I'm not sure we need to make that performance optimization right now.

Keeping the queries for tickets, clients, pets, etc. separated makes them more maintainable. If we want to transform or filter certain fields for a Client, we can do that in one place if we keep the queries separated. Else, every scenario that needs to fetch a user would have to duplicate the user-specific parts of the query.

Reply:
An entity does not always align to a single table.  In this case, we have defined service-request as being comprised of state info about the request AND a client.  Our application has mapped the service-request onto two tables.

Reply:
Jeff, do you mean since service_request table rows basically 3-4 columns of just id to specific table. Did you mean you want us to attempt to destruct those out existing ticket, and pass them on compounding joins select? we can do like select(app_constant('app_constant_id),clients('clients_id'),animals('animal_id'),employee('emp_id')? 

What I am asking is probably way off if it is ignore this comment lol

Reply:
I understand what you're staring @jnakaso, but I don't think we've formally defined these entities. And I thought we were trying to keep things simple so they are easy to maintain and understand for future developers. IMO a 1-to-1 mapping of entities to tables is the easiest to understand.

Reply:
> Jeff, do you mean since service_request table rows basically 3-4 columns of just id to specific table. Did you mean you want us to attempt to destruct those out existing ticket, and pass them on compounding joins select? we can do like select(app_constant('app_constant_id),clients('clients_id'),animals('animal_id'),employee('emp_id')? 
> 
> 
> 
> What I am asking is probably way off if it is ignore this comment lol

Yes I think Jeff is simply suggesting doing the joins in the query here. I'm advocating for keeping the entities separate, as if they came from separate data sources altogether. I think this separation, while less performant, makes more sense for a small scale solution like ours. I really want to avoid anyone having to read a query to understand what data they are getting. IMO they should be able to tell from the function name what it returns. To my understanding a ticket entity does not have a first name like a client entity does.

Reply:
I wonder if we need some kind of DB wrapper so we can separate the responsibilities of the code. It feels like this conversation about efficiency vs adhering to REST is not really relevant to this change.

My opinion is that we keep the code as written here and then we can create a clean up ticket which can abstract this discussion to a different class

Reply:
Hey I took a look at the network requests and it looks concerning:

![Jun-08-2024 15-02-01](https://github.com/digitalaidseattle/seattle-humane-app/assets/25556678/2a7f96c8-1774-457e-acd5-1931787b1da0)

I didn't see this behavior on the `sprint-4` branch

@jnakaso was this your concern here?